sklearn/utils/_mocking.py

```diff
@@ -70,10 +70,12 @@ class CheckingClassifier(ClassifierMixin, BaseEstimator):
     ----------
     check_y, check_X : callable, default=None
         The callable used to validate `X` and `y`. These callable should return
-        a bool where `False` will trigger an `AssertionError`.
+        a bool where `False` will trigger an `AssertionError`. If `None`, the
+        data is not validated. Default is `None`.
 
     check_y_params, check_X_params : dict, default=None
-        The optional parameters to pass to `check_X` and `check_y`.
+        The optional parameters to pass to `check_X` and `check_y`. If `None`,
+        an empty set is passed to the check_y/check_X callable.
```

Comment:
Nit: `{}` is technically an empty dict, but we need to specify the data type. I think it's clear enough to say "no parameters are passed in":

```suggestion
        The optional parameters to pass to `check_X` and `check_y`. If `None`,
        then no parameters are passed in.
```